index.js

```diff
@@ -62,7 +62,7 @@ module.exports = class MongooseTrailpack extends DatastoreTrailpack {
         schema.statics = model.statics
         schema.methods = model.methods
 
-        model.onSchema(schema)
+        model.onSchema(schema, this.app)
```

Comment:
Sorry, I think I didn't explained well
I'm talking about moving the onSchema method in `config` object, somethings like this:

```
'use strict'
...
module.exports = {
  config:   (app, Mongoose) => {
    return {
      schema:   {
        timestamps: true,
        versionKey: false
      },
      methods:  {
        toJSON: function() {
          const user = this.toObject()
          user.id    = user._id
          delete user.password

          return user
        }
      },
      // REFACTORED HERE
      onSchema: function(app, schema) {
        schema.pre('save', function(next) {
          hashPassword(app.config.passport.bcrypt, this, next)
        })
      }
    }
  },
  schema: (app, Mongoose){
    ...
  }
}
```